lib/rouge/lexer.rb

```diff
@@ -423,15 +423,24 @@ def reset!
     #
     # @note The use of `opts` has been deprecated. A warning is issued if run
     #   with `$VERBOSE` set to true.
-    def lex(string, opts={}, &b)
-      unless opts.nil?
-        warn 'The use of opts with Lexer.lex is deprecated' if $VERBOSE
+    def lex(string, opts=nil, &b)
+      if opts
+        if (opts.keys - [:continue]).size > 0
```

Comment:
You can also use [`Array#empty?`](https://ruby-doc.org/core-2.0.0/Array.html#method-i-empty-3F) here:
```suggestion
        unless (opts.keys - [:continue]).empty?
```